test/L2TokenBridge.t.sol

```diff
@@ -206,4 +214,78 @@ contract L2TokenBridgeTest is DssTest {
         assertEq(l2Token.balanceOf(address(0xced)), balanceBefore + 100 ether);
         assertEq(l2Token.totalSupply(), supplyBefore + 100 ether);
     }
+
+    function testDeployWithUpgradesLib() public {
+        Options memory opts;
+        if (!validate) {
+            opts.unsafeSkipAllChecks = true;
+        } else {
+            opts.unsafeAllow = 'state-variable-immutable,constructor';
+        }
+        opts.constructorData = abi.encode(otherBridge, address(messenger));
+
+        vm.expectEmit(true, true, true, true);
+        emit Rely(address(this));
+        address proxy = Upgrades.deployUUPSProxy(
+            "out/L2TokenBridge.sol/L2TokenBridge.json",
+            abi.encodeCall(L2TokenBridge.initialize, ()),
+            opts
+        );
+        assertEq(L2TokenBridge(proxy).version(), "1");
+        assertEq(L2TokenBridge(proxy).wards(address(this)), 1);
+    }
+
+    function testUpgrade() public {
+        address newImpl = address(new L2TokenBridgeV2Mock());
+        vm.expectEmit(true, true, true, true);
+        emit UpgradedTo("2");
+        bridge.upgradeToAndCall(newImpl, abi.encodeCall(L2TokenBridgeV2Mock.reinitialize, ()));
+
+        assertEq(bridge.getImplementation(), newImpl);
+        assertEq(bridge.version(), "2");
+        assertEq(bridge.wards(address(this)), 1); // still a ward
+    }
+
+    function testUpgradeWithUpgradesLib() public {
+        address implementation1 = bridge.getImplementation();
+
+        Options memory opts;
+        if (!validate) {
+            opts.unsafeSkipAllChecks = true;
+        } else {
+            opts.referenceContract = "out/L2TokenBridge.sol/L2TokenBridge.json";
+            opts.unsafeAllow = 'constructor';
+        }
+
+        vm.expectEmit(true, true, true, true);
+        emit UpgradedTo("2");
+        Upgrades.upgradeProxy(
+            address(bridge),
+            "out/L2TokenBridgeV2Mock.sol/L2TokenBridgeV2Mock.json",
+            abi.encodeCall(L2TokenBridgeV2Mock.reinitialize, ()),
+            opts
+        );
+
+        address implementation2 = bridge.getImplementation();
+        assertTrue(implementation1 != implementation2);
+        assertEq(bridge.version(), "2");
+        assertEq(bridge.wards(address(this)), 1); // still a ward
+    }
+
+    function testUpgradeUnauthed() public {
```

Comment:
Same here, we could remove this test adding `bridge.upgradeToAndCall.selector` to `testAuthModifiers`.

Reply:
Done